internal/storage/bucket_handle_test.go

```diff
@@ -440,13 +450,9 @@ func (testSuite *BucketHandleTest) TestListObjectMethodWithPrefixObjectExist() {
 		})
 
 	assert.Nil(testSuite.T(), err)
-	assert.Equal(testSuite.T(), 4, len(obj.Objects))
-	assert.Equal(testSuite.T(), 1, len(obj.CollapsedRuns))
-	assert.Equal(testSuite.T(), TestObjectRootFolderName, obj.Objects[0].Name)
-	assert.Equal(testSuite.T(), TestObjectSubRootFolderName, obj.Objects[1].Name)
-	assert.Equal(testSuite.T(), TestObjectName, obj.Objects[2].Name)
+	assert.Equal(testSuite.T(), []string{TestObjectRootFolderName, TestObjectSubRootFolderName, TestObjectName, TestGzipObjectName}, objectsToObjectNames(obj.Objects))
```

Comment:
> Does assert takes care of panics in case of length mismatch?

Yes. It doesn't panic and handles length mismatches well by gracefully failing on them.